connectivity/drivers/cellular/TELIT/ME310/TELIT_ME310_CellularStack.cpp

```diff
@@ -440,7 +436,7 @@ nsapi_size_or_error_t TELIT_ME310_CellularStack::socket_recvfrom_impl(CellularSo
                 // read() should not fail
                 success = false;
             }
-        } else if (timer.read_ms() < ME310_SOCKET_TIMEOUT) {
+        } else if (std::chrono::duration_cast<std::chrono::milliseconds>(timer.elapsed_time()) < std::chrono::milliseconds(ME310_SOCKET_TIMEOUT)) {
```

Comment:
I think it might be better to change the definition of `ME310_SOCKET_TIMEOUT` to be of chrono type and then change the code that references it.

Reply:
since ME310_SOCKET_TIMEOUT is a define it has no type... 
do you mean to make it a constexpr? I did not see constexpr used very often in Mbed code I thought it does not comply with coding guidelines

Reply:
I committed the changes you suggested

Reply:
See an example of a chrono macro here: https://github.com/ARMmbed/mbed-os/blob/master/connectivity/cellular/source/framework/device/CellularStateMachine.cpp#L30

Reply:
thank you for the example!
is it ok to use a constepr in this case? i myself like them more because they know namesspaces and are strictly typed. 
or does this not comply with mbed-code style

Reply:
I believe both are acceptable, the advantage of the constexpr is that it can be moved easily to an header file whereas the macro requires a using directive of the `std::chrono_literals` namespace to work.

Reply:
You indeed can't define something as `1s` in a header because of the user-defined-literal requiring the `using` directive. I find that really annoying.

But you can do `#define XXX_TIMEOUT std::chrono::seconds{1}` - we're using that form in a few headers.

I'd be equally fine with having it as a `constexpr` definition.

But usually time constants don't need to be in headers anyway - they're normally implementation details. Main exception I've found is default parameters, like having a zero-timeout default.